ortools/linear_solver/xpress_interface.cc

```diff
@@ -1589,6 +1622,27 @@ void XpressInterface::SetLpAlgorithm(int value) {
   }
 }
 
+void XpressInterface::SetStartingLpBasis(
+    const std::vector<int>& variable_statuses,
+    const std::vector<int>& constraint_statuses) {
+
+  initRstat = constraint_statuses.data();
+  initCstat = variable_statuses.data();
```

Comment:
This implementation will crash because of ownership issues.

Typical scenario
```
{
std::vector<int> v:
std::vector<int> c;
solver->SetStartingLpBasis(v, c);
}
// v and c are released, c.data() and v.data() point to an invalid addresses
solver->Solve(); // SEGFAULT
```
You want to

- Avoid dereferencing invalid pointers
- Avoid copies

This is a good use case for `std::move` and [move-semantics](https://www.cprogramming.com/c++11/rvalue-references-and-move-semantics-in-c++11.html).

```cpp
void XpressInterface::SetStartingLpBasis(std::vector<int>&& variable_statuses,
    std::vector<int>&& constraint_statuses)
{
//
}
```
```cpp
{
std::vectror<int> v;
solver->SetStartingLpBasis(v, c); // solver owns v & c, they can be released
}
solver->Solve(); // OK
```
You need to keep statuses as `std::vectors`, not pointers.